pe_exceptions/core/pe_base_exception.hpp

```diff
@@ -0,0 +1,30 @@
+#ifndef PE_BASE_EXCEPTION_HPP
+#define PE_BASE_EXCEPTION_HPP
+
+#include <fstream>
+#include <stdexcept>
+#include <string_view>
+
+namespace peach::detail
+{
+  class pe_base_exception : virtual public std::runtime_error
+  {
+  public:
+    pe_base_exception( ) = default;
+
+    // clang-format off
+    pe_base_exception( const pe_base_exception& ) noexcept = default;
+    pe_base_exception( pe_base_exception&& ) noexcept = delete;
+
+    pe_base_exception& operator=( pe_base_exception&& ) = delete;
+    pe_base_exception& operator=( const pe_base_exception& ) = delete;
```

Comment:
Why delete these?

Reply:
Why would you allow exceptions to be moved or copied? I mean I guess move assignment is ok but I dont see any actual usecase for such

Reply:
I would say if its okay to move it, allow moving it. You never know how someone might want to use/extend the library.

Reply:
fixed in 79cab40